### PR TITLE
Add typescript support

### PR DESCRIPTION
**Pull request by devnoot**
This adds TypeScript support per #3

### CODE REVIEW

**Comment by esthor** [comment]
Thanks for adding this @devnoot. Looks good from the diff. I haven't had a chance to build and use it yet. Did it build and work fine for you?

**Comment by devnoot** [comment]
@esthor I haven't, but I should have some time this weekend to test

**Comment by esthor** [comment]
Awesome. Thank you. I'm really looking forward to having this in the library and cutting a release with typescript types 🤘

File: types/react-native-swipeable-list/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  import React from 'react';
+  import {FlatListProps} from 'react-native';
+
+  interface SwipeableFlatListProps<ItemT> extends FlatListProps<ItemT> {
+    maxSwipeDistance?: number;
+    shouldBounceOnMount?: boolean;
+    renderQuickActions: ({index: number, item: ItemT}) => React.ReactNode;
+  }
+
+  class SwipeableFlatList<ItemT = any> extends React.Component<
```
Comment (esthor) [inline types/react-native-swipeable-list/index.d.ts:15]: Is it possible to tighten up the typing here instead of an `any`?

**Comment by esthor** [comment]
How can I help?

**Pull request closed without merging**